tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -933,6 +933,15 @@ func (az *AzureProvider) destroyDisk(ctx context.Context, vmInfo compute.Virtual
 	return
 }
 
+// DestroyWindowsVM destroys the given Windows VM
+func (az *AzureProvider) DestroyWindowsVM(vmName string) error {
+	ctx := context.Background()
+	if err := az.destroyInstance(ctx, vmName); err != nil {
```

Comment:
This is not sufficient for azure. You will have to do what is being done in `DestroyWindowsVMs()` otherwise you will leak those other resources. Please make a common helper that you call from both the places.

Reply:
Thanks for catching, I've added it now.